Server/src/main/java/net/simon987/server/event/ObjectDeathEvent.java

```diff
@@ -0,0 +1,18 @@
+package net.simon987.server.event;
+
+/**
+ * Event dispatched by a GameObject who has needed callbacks on death
+ */
+public class ObjectDeathEvent extends GameEvent {
+    /**
+     * The GameObject type ID of object that init this event
+     */
+    private int sourceObjectId;
```

Comment:
It feels a bit weird to have to store the object type id, but that's my fault. I think that an abstract GameObject.getTypeId() method could be added, which would avoid either storing the id or comparing the MAP_INFO mask like I did in [HarvestTask#40](https://github.com/simon987/Much-Assembly-Required/blob/9e402fe8a12b2cd175ccef463732d2e88b64e271/Plugin%20NPC/src/main/java/net/simon987/npcplugin/HarvestTask.java#L40)    
You can keep it like this for now and I'll update it as well as the other case in the NPC Plugin later on. (The <GameObject>.ID constants need to be renamed to TYPE_ID to avoid confusion with GameObject.getObjectId() as well)

Reply:
Maybe it's ok to cast event.getSource() to `GameObject` if `ObjectDeathEvent` implies it'll only be an event for `GameObject`? Then we might be able to access the `ID` attribute which make `sourceObjectId` redundant.